Makefile

```diff
@@ -59,11 +60,17 @@ test: $(test_exes) $(test_dir)/end_to_end_test.exe
 $(test_dir)/%.exe: $(test_dir)/%.o
 	$(CXX) $(CXXFLAGS) $< -o $@
 
+$(test_dir)/multiple_files.exe: $(test_dir)/end_to_end_test.o $(test_dir)/no_main.o
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+$(test_dir)/%.run: $(test_dir)/%.exe
+	$(call run_and_diff,$<,$*)
+
 # Params:
 # 1. executable plus args
 # 2. base for .correct.txt and .out.txt files
 define run_and_diff
-	-./$1 > $(test_dir)/$2.out.txt 2>&1;
+	-./$1 | sed "s/0x[0-9a-fA-F]*/<HEX VALUE>/g" > $(test_dir)/$2.out.txt 2>&1;
```

Comment:
@amirkamil What is this wizardry?

Reply:
`test_special_asserts.cpp` asserts that two pointers are equal, which ends up being false. The framework prints out the pointer values, but the values differ between different runs. This `sed` command replaces hex values in the output with a standard string.